sdk/storage/Azure.Storage.Webjobs.Extensions/src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Bindings/StorageBlobContainerExtensions.cs

```diff
@@ -35,24 +37,38 @@ internal static class StorageBlobContainerExtensions
             }
         }
 
-        private static async Task<ICloudBlob> GetExistingOrNewBlockBlobReferenceAsync(CloudBlobContainer container,
+        public static async Task<(BlobBaseClient, BlobProperties)> GetBlobReferenceFromServerAsync(this BlobContainerClient container, string blobName, CancellationToken cancellationToken = default)
+        {
+            BlobProperties blobProperties = await container.GetBlobClient(blobName).GetPropertiesAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            switch (blobProperties.BlobType)
+            {
+                case BlobType.Append:
+                    return (container.GetAppendBlobClient(blobName), blobProperties);
+                case BlobType.Block:
+                    return (container.GetBlockBlobClient(blobName), blobProperties);
+                case BlobType.Page:
+                    return (container.GetPageBlobClient(blobName), blobProperties);
+                default:
+                    throw new InvalidOperationException();
+            }
+        }
+
+        private static async Task<BlobBaseClient> GetExistingOrNewBlockBlobReferenceAsync(BlobContainerClient container,
             string blobName, CancellationToken cancellationToken)
         {
             try
             {
-                return await container.GetBlobReferenceFromServerAsync(blobName, cancellationToken).ConfigureAwait(false);
+                return (await container.GetBlobReferenceFromServerAsync(blobName, cancellationToken).ConfigureAwait(false)).Item1;
```

Comment:
What is the type of container here?

Reply:
`Item1` is declared `BlobBaseClient`. However, reference would be one of specialized clients in runtime - resolved by calling get properties as a first step inside extension.